arbor/s_expr.cpp

```diff
@@ -281,7 +281,7 @@ class lexer {
         }
 
         // test if the symbol matches a keyword
-        auto it = keyword_to_tok.find(symbol.c_str());
+        auto it = keyword_to_tok.find(symbol);
```

Comment:
There are two kinds of people: those that append `.c_str()` gratuitously and those that really need it. How can PVSS tell which it is?

Reply:
`Find` takes a `const std::string&` as `std::string` is the key type?!

Reply:
`.c_str()` isn't a `std::string`?

Reply:
It's a `const char*`, like in C. If `std::string` where a `std::vector<char>`, `c_str` would be like `data`, returning the underlying buffer by pointer.

Reply:
Yes, but that's not what I meant. I guess the answer to my question is: yes, in this case PVSS knows how `symbol` is used and `c_str()` is not required for that use.

Reply:
It knows since it can see this
```
std::map<K, V>::find(const K&)
```
and here `K=std::string`. We are calling
```
map.find(string.c_str())
```
so the temporary C string is converted back to a `std::string`.
That in turn is not possible without copying the bytes, so
1. not needed
2. wasteful